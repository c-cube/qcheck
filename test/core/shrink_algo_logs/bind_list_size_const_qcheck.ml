open QCheck

let bind_pair_list_size =
  let shrink (_l,xs) =
    Iter.map (fun xs' -> (List.length xs',xs')) Shrink.(list ~shrink:int xs) in
  Test.make ~name:"bind list_size constant"
    (make ~shrink
       Gen.(int_bound 1000 >>= fun len ->
            list_size (return len) (int_bound 1000) >>= fun xs -> return (len,xs)))
    (Log.shrinks Print.(pair int (list int)) @@
       fun (len,xs) -> let len' = List.length xs in len=len' && len' < 4)
;;
try Test.check_exn bind_pair_list_size with Test.Test_fail _ -> ()
