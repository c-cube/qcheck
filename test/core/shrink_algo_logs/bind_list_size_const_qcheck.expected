fails (385, [666; 863; 235; 895; 235; 241; 408; 927; 32; 2; 606; 896; 513; 762; 910; 973; 86; 599; 574; 612; 0; 267; 567; 196; 495; 239; 353; 864; 5; 627; 100; 121; 910; 438; 880; 536; 212; 843; 281; 323; 93; 868; 505; 649; 719; 905; 344; 708; 209; 351; 633; 22; 634; 525; 691; 556; 953; 805; 0; 844; 43; 696; 810; 707; 906; 907; 392; 2; 35; 166; 620; 377; 392; 212; 993; 144; 625; 42; 685; 926; 300; 711; 32; 297; 591; 571; 457; 187; 538; 110; 152; 339; 826; 279; 876; 720; 457; 463; 648; 718; 427; 302; 184; 281; 216; 90; 989; 505; 927; 262; 98; 905; 431; 68; 705; 843; 864; 518; 969; 186; 596; 97; 362; 493; 743; 309; 357; 275; 989; 68; 168; 256; 288; 420; 831; 399; 281; 110; 746; 887; 713; 92; 344; 345; 644; 218; 960; 515; 534; 743; 507; 329; 485; 754; 553; 172; 177; 758; 583; 809; 789; 26; 929; 183; 551; 159; 731; 92; 424; 671; 883; 500; 978; 387; 847; 264; 618; 751; 944; 573; 116; 606; 692; 312; 280; 141; 498; 107; 996; 682; 807; 331; 665; 74; 991; 361; 133; 954; 632; 403; 583; 346; 556; 963; 632; 43; 624; 318; 491; 406; 827; 76; 923; 904; 168; 587; 528; 867; 192; 337; 176; 766; 619; 167; 78; 694; 124; 21; 737; 944; 784; 480; 260; 543; 872; 30; 681; 962; 138; 848; 776; 622; 916; 680; 684; 46; 712; 510; 141; 313; 236; 120; 357; 686; 610; 277; 72; 22; 244; 754; 341; 659; 314; 661; 812; 337; 167; 384; 388; 119; 899; 25; 891; 51; 96; 49; 554; 922; 691; 976; 774; 116; 770; 117; 787; 133; 298; 745; 899; 530; 865; 448; 747; 146; 110; 749; 898; 17; 577; 620; 652; 920; 972; 169; 544; 259; 517; 420; 49; 475; 123; 331; 307; 866; 865; 889; 72; 166; 549; 76; 445; 296; 814; 769; 469; 106; 234; 715; 504; 12; 680; 298; 878; 212; 24; 611; 951; 68; 420; 915; 58; 156; 203; 967; 217; 728; 641; 981; 705; 161; 671; 322; 520; 727; 930; 512; 25; 930; 305; 96; 441; 430; 166; 269; 569; 967; 120; 347; 644; 140; 254; 241; 622; 567; 422; 283; 788; 249; 15; 124; 90; 311; 413; 572; 720])
fails (192, [74; 991; 361; 133; 954; 632; 403; 583; 346; 556; 963; 632; 43; 624; 318; 491; 406; 827; 76; 923; 904; 168; 587; 528; 867; 192; 337; 176; 766; 619; 167; 78; 694; 124; 21; 737; 944; 784; 480; 260; 543; 872; 30; 681; 962; 138; 848; 776; 622; 916; 680; 684; 46; 712; 510; 141; 313; 236; 120; 357; 686; 610; 277; 72; 22; 244; 754; 341; 659; 314; 661; 812; 337; 167; 384; 388; 119; 899; 25; 891; 51; 96; 49; 554; 922; 691; 976; 774; 116; 770; 117; 787; 133; 298; 745; 899; 530; 865; 448; 747; 146; 110; 749; 898; 17; 577; 620; 652; 920; 972; 169; 544; 259; 517; 420; 49; 475; 123; 331; 307; 866; 865; 889; 72; 166; 549; 76; 445; 296; 814; 769; 469; 106; 234; 715; 504; 12; 680; 298; 878; 212; 24; 611; 951; 68; 420; 915; 58; 156; 203; 967; 217; 728; 641; 981; 705; 161; 671; 322; 520; 727; 930; 512; 25; 930; 305; 96; 441; 430; 166; 269; 569; 967; 120; 347; 644; 140; 254; 241; 622; 567; 422; 283; 788; 249; 15; 124; 90; 311; 413; 572; 720])
fails (96, [530; 865; 448; 747; 146; 110; 749; 898; 17; 577; 620; 652; 920; 972; 169; 544; 259; 517; 420; 49; 475; 123; 331; 307; 866; 865; 889; 72; 166; 549; 76; 445; 296; 814; 769; 469; 106; 234; 715; 504; 12; 680; 298; 878; 212; 24; 611; 951; 68; 420; 915; 58; 156; 203; 967; 217; 728; 641; 981; 705; 161; 671; 322; 520; 727; 930; 512; 25; 930; 305; 96; 441; 430; 166; 269; 569; 967; 120; 347; 644; 140; 254; 241; 622; 567; 422; 283; 788; 249; 15; 124; 90; 311; 413; 572; 720])
fails (48, [68; 420; 915; 58; 156; 203; 967; 217; 728; 641; 981; 705; 161; 671; 322; 520; 727; 930; 512; 25; 930; 305; 96; 441; 430; 166; 269; 569; 967; 120; 347; 644; 140; 254; 241; 622; 567; 422; 283; 788; 249; 15; 124; 90; 311; 413; 572; 720])
fails (24, [430; 166; 269; 569; 967; 120; 347; 644; 140; 254; 241; 622; 567; 422; 283; 788; 249; 15; 124; 90; 311; 413; 572; 720])
fails (12, [567; 422; 283; 788; 249; 15; 124; 90; 311; 413; 572; 720])
fails (6, [124; 90; 311; 413; 572; 720])
holds (3, [413; 572; 720])
holds (3, [124; 572; 720])
holds (3, [124; 90; 720])
holds (3, [124; 90; 311])
fails (5, [90; 311; 413; 572; 720])
holds (2, [572; 720])
holds (2, [90; 720])
holds (2, [90; 311])
fails (4, [311; 413; 572; 720])
holds (2, [572; 720])
holds (2, [311; 720])
holds (2, [311; 413])
holds (3, [413; 572; 720])
holds (3, [311; 572; 720])
holds (3, [311; 413; 720])
holds (3, [311; 413; 572])
fails (4, [156; 413; 572; 720])
holds (2, [572; 720])
holds (2, [156; 720])
holds (2, [156; 413])
holds (3, [413; 572; 720])
holds (3, [156; 572; 720])
holds (3, [156; 413; 720])
holds (3, [156; 413; 572])
fails (4, [78; 413; 572; 720])
holds (2, [572; 720])
holds (2, [78; 720])
holds (2, [78; 413])
holds (3, [413; 572; 720])
holds (3, [78; 572; 720])
holds (3, [78; 413; 720])
holds (3, [78; 413; 572])
fails (4, [39; 413; 572; 720])
holds (2, [572; 720])
holds (2, [39; 720])
holds (2, [39; 413])
holds (3, [413; 572; 720])
holds (3, [39; 572; 720])
holds (3, [39; 413; 720])
holds (3, [39; 413; 572])
fails (4, [20; 413; 572; 720])
holds (2, [572; 720])
holds (2, [20; 720])
holds (2, [20; 413])
holds (3, [413; 572; 720])
holds (3, [20; 572; 720])
holds (3, [20; 413; 720])
holds (3, [20; 413; 572])
fails (4, [10; 413; 572; 720])
holds (2, [572; 720])
holds (2, [10; 720])
holds (2, [10; 413])
holds (3, [413; 572; 720])
holds (3, [10; 572; 720])
holds (3, [10; 413; 720])
holds (3, [10; 413; 572])
fails (4, [5; 413; 572; 720])
holds (2, [572; 720])
holds (2, [5; 720])
holds (2, [5; 413])
holds (3, [413; 572; 720])
holds (3, [5; 572; 720])
holds (3, [5; 413; 720])
holds (3, [5; 413; 572])
fails (4, [3; 413; 572; 720])
holds (2, [572; 720])
holds (2, [3; 720])
holds (2, [3; 413])
holds (3, [413; 572; 720])
holds (3, [3; 572; 720])
holds (3, [3; 413; 720])
holds (3, [3; 413; 572])
fails (4, [2; 413; 572; 720])
holds (2, [572; 720])
holds (2, [2; 720])
holds (2, [2; 413])
holds (3, [413; 572; 720])
holds (3, [2; 572; 720])
holds (3, [2; 413; 720])
holds (3, [2; 413; 572])
fails (4, [1; 413; 572; 720])
holds (2, [572; 720])
holds (2, [1; 720])
holds (2, [1; 413])
holds (3, [413; 572; 720])
holds (3, [1; 572; 720])
holds (3, [1; 413; 720])
holds (3, [1; 413; 572])
fails (4, [0; 413; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 413])
holds (3, [413; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 413; 720])
holds (3, [0; 413; 572])
fails (4, [0; 207; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 207])
holds (3, [207; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 207; 720])
holds (3, [0; 207; 572])
fails (4, [0; 104; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 104])
holds (3, [104; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 104; 720])
holds (3, [0; 104; 572])
fails (4, [0; 52; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 52])
holds (3, [52; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 52; 720])
holds (3, [0; 52; 572])
fails (4, [0; 26; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 26])
holds (3, [26; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 26; 720])
holds (3, [0; 26; 572])
fails (4, [0; 13; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 13])
holds (3, [13; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 13; 720])
holds (3, [0; 13; 572])
fails (4, [0; 7; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 7])
holds (3, [7; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 7; 720])
holds (3, [0; 7; 572])
fails (4, [0; 4; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 4])
holds (3, [4; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 4; 720])
holds (3, [0; 4; 572])
fails (4, [0; 2; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 2])
holds (3, [2; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 2; 720])
holds (3, [0; 2; 572])
fails (4, [0; 1; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 1])
holds (3, [1; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 1; 720])
holds (3, [0; 1; 572])
fails (4, [0; 0; 572; 720])
holds (2, [572; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 572; 720])
holds (3, [0; 572; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 572])
fails (4, [0; 0; 286; 720])
holds (2, [286; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 286; 720])
holds (3, [0; 286; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 286])
fails (4, [0; 0; 143; 720])
holds (2, [143; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 143; 720])
holds (3, [0; 143; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 143])
fails (4, [0; 0; 72; 720])
holds (2, [72; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 72; 720])
holds (3, [0; 72; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 72])
fails (4, [0; 0; 36; 720])
holds (2, [36; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 36; 720])
holds (3, [0; 36; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 36])
fails (4, [0; 0; 18; 720])
holds (2, [18; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 18; 720])
holds (3, [0; 18; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 18])
fails (4, [0; 0; 9; 720])
holds (2, [9; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 9; 720])
holds (3, [0; 9; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 9])
fails (4, [0; 0; 5; 720])
holds (2, [5; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 5; 720])
holds (3, [0; 5; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 5])
fails (4, [0; 0; 3; 720])
holds (2, [3; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 3; 720])
holds (3, [0; 3; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 3])
fails (4, [0; 0; 2; 720])
holds (2, [2; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 2; 720])
holds (3, [0; 2; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 2])
fails (4, [0; 0; 1; 720])
holds (2, [1; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 1; 720])
holds (3, [0; 1; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 1])
fails (4, [0; 0; 0; 720])
holds (2, [0; 720])
holds (2, [0; 720])
holds (2, [0; 0])
holds (3, [0; 0; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 720])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 360])
holds (2, [0; 360])
holds (2, [0; 360])
holds (2, [0; 0])
holds (3, [0; 0; 360])
holds (3, [0; 0; 360])
holds (3, [0; 0; 360])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 180])
holds (2, [0; 180])
holds (2, [0; 180])
holds (2, [0; 0])
holds (3, [0; 0; 180])
holds (3, [0; 0; 180])
holds (3, [0; 0; 180])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 90])
holds (2, [0; 90])
holds (2, [0; 90])
holds (2, [0; 0])
holds (3, [0; 0; 90])
holds (3, [0; 0; 90])
holds (3, [0; 0; 90])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 45])
holds (2, [0; 45])
holds (2, [0; 45])
holds (2, [0; 0])
holds (3, [0; 0; 45])
holds (3, [0; 0; 45])
holds (3, [0; 0; 45])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 23])
holds (2, [0; 23])
holds (2, [0; 23])
holds (2, [0; 0])
holds (3, [0; 0; 23])
holds (3, [0; 0; 23])
holds (3, [0; 0; 23])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 12])
holds (2, [0; 12])
holds (2, [0; 12])
holds (2, [0; 0])
holds (3, [0; 0; 12])
holds (3, [0; 0; 12])
holds (3, [0; 0; 12])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 6])
holds (2, [0; 6])
holds (2, [0; 6])
holds (2, [0; 0])
holds (3, [0; 0; 6])
holds (3, [0; 0; 6])
holds (3, [0; 0; 6])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 3])
holds (2, [0; 3])
holds (2, [0; 3])
holds (2, [0; 0])
holds (3, [0; 0; 3])
holds (3, [0; 0; 3])
holds (3, [0; 0; 3])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 2])
holds (2, [0; 2])
holds (2, [0; 2])
holds (2, [0; 0])
holds (3, [0; 0; 2])
holds (3, [0; 0; 2])
holds (3, [0; 0; 2])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 1])
holds (2, [0; 1])
holds (2, [0; 1])
holds (2, [0; 0])
holds (3, [0; 0; 1])
holds (3, [0; 0; 1])
holds (3, [0; 0; 1])
holds (3, [0; 0; 0])
fails (4, [0; 0; 0; 0])
holds (2, [0; 0])
holds (2, [0; 0])
holds (2, [0; 0])
holds (3, [0; 0; 0])
holds (3, [0; 0; 0])
holds (3, [0; 0; 0])
holds (3, [0; 0; 0])
